syntax = "proto3";

package rsync;

service Rsync {
  rpc PreCalcu (FileHead) returns (RsyncReply) {}
  rpc CalcuDiff (FileInfo) returns (RsyncReply) {}
}

message FileHead {
  string filename = 1;
  int32 filesize = 2;
  int32 chunksize = 3;
}

message FileInfo {
  repeated FileChunkInfo chunkinfo = 1;
  string filename = 2;
}

message FileChunkInfo {
	int32 chunkid = 1;
	int32 checksum1 = 2;
	bytes checksum2 = 3;
}

message RsyncReply {
	bool success = 1;
}